### PR TITLE
For #17872, #18032: Remove disabled color for three dot menu icons and add report site issue

### PR DESCRIPTION
**Pull request by eliserichards**
For #17872 (icons), for #18032 (report site issue)

This makes the three dot menu icons visible again (per design decision). This also adds the "Report site issue" item in nightly/debug.

Confirmed colors in: 
- [x] light mode, normal browsing (see below)
- [x] light mode, private browsing (no screenshots)
- [x] dark mode, normal browsing (see below)
- [x] dark mode, private browsing (no screenshots)

### Screenshots

 

Open in app & reader view appearance: 

Report site issue: (FYI there's an open bug for making the lightbulb icon look better.. I'll link that here)

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -370,56 +369,56 @@ class DefaultToolbarMenu(
     private val newCoreMenuItems by lazy {
         val newTabItem = BrowserMenuImageText(
             context.getString(R.string.library_new_tab),
-            R.drawable.ic_bookmark_filled,
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:373]: Good catch!

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 2]
```diff
@@ -186,7 +185,7 @@ class DefaultToolbarMenu(
         appLink(session.content.url).hasExternalApp()
     } ?: false
 
-    private fun shouldShowReaderAppearance(): Boolean = selectedSession?.let {
+    private fun shouldShowReaderViewCustomization(): Boolean = selectedSession?.let {
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:188]: nit: this function renaming seems related to the menu refactoring instead of to this PR (i.e. adding 3-dot menu icons). Let's try to keep unrelated changes separate.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 3]
```diff
@@ -434,41 +433,59 @@ class DefaultToolbarMenu(
...
+        val customizeReaderView = BrowserMenuImageText(
             label = context.getString(R.string.browser_menu_customize_reader_view),
-            textColorResource = primaryTextColor()
+            imageResource = R.drawable.ic_readermode_appearance,
+            iconTintColorResource = primaryTextColor()
         ) {
             onItemTapped.invoke(ToolbarMenu.Item.CustomizeReaderView)
         }
 
+        val openInApp = BrowserMenuHighlightableItem(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:444]: Similar nit as above: Why are we adding this here? Seems related to a different PR.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 4]
```diff
@@ -370,56 +369,56 @@ class DefaultToolbarMenu(
...
             R.drawable.ic_addons_extensions,
-            disabledTextColor()
+            primaryTextColor()
         ) {
             onItemTapped.invoke(ToolbarMenu.Item.AddonsManager)
         }
 
-        val syncedTabsItem = BrowserMenuImageText(
-            context.getString(R.string.library_synced_tabs),
+        val syncSignIn = BrowserMenuImageText(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:410]: Same nit as above: Let's leave this as part of the other PR. This is a half change as well because we need to update the `onItemTapped.invoke(ToolbarMenu.Item.SyncedTabs)` as well.

**Review by jonalmeida — CHANGES_REQUESTED** [review]
Looks good; let's remove some changes that are not part of this PR before landing!

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 2 above)
Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:188]: The reader view PR was already merged, and this rename (from "Appearance" to "Cutomization") was missed. I would like to keep it in this PR so I don't have to open a separate one just for a rename

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 4 above)
Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:410]: I agree. The name change from synced tabs to sync sign in shouldn't be part of this card. Creating the invocation for ToolbarMenu.Item.SyncSignIn (or whatever it will be called) and changing that name will be part of https://github.com/mozilla-mobile/fenix/issues/17798

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 3 above)
Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:444]: It was taken out as a part of the original design, and the updated designs add it back in. I could open a separate PR but that seemed wasteful

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:444]: PRs are cheap. 🙂 

It's a good rule of thumb to keep isolated work separately so there is context when someone needs to debug why a change was added to a separate issue. Smaller PRs are also easier for the reviewer.

How about separating this into a different commit so it doesn't fall under "Remove disabled color for icons"?

Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:444]: Sure, I'll tag you ;)

**Review by jonalmeida — APPROVED** [review]
LGTM!
